Replicator/tests/SG.cc

```diff
@@ -116,6 +117,9 @@ bool SG::deleteUser(const string& username) const {
 
 bool SG::assignUserChannel(const std::string& username, const std::vector<C4CollectionSpec>& collectionSpecs,
                            const std::vector<std::string>& channelIDs) const {
+    if ( getServerName() < "Couchbase Sync Gateway/3.1" ) return assignUserChannelOld(username, channelIDs);
```

Comment:
Can you confirm this is intentional and make a comment saying why it works?  Is this comparing the bytes of the two slices by ASCII value?

Reply:
I think so. I've done a lot of testing with it, so I know it works, but I don't entirely know what it's doing. I'll try and look into it.

Reply:
Yeah, under the hood, this does a `memcmp` of the two slices. So I'm gonna say that'll be the ASCII bytes. Maybe some weird scenarios where this would say the version is correct when it isn't / vice versa, but I think it'll do for now. I might add a log call in, to say ~"assigning user channels for SG < 3.1", so if it becomes a problem at any point it can be found easily.

Reply:
It will be ok until we get to version 3.10, in which case it will sort before 3.2 ;)